tools/k-rule-find/main.cpp

```diff
@@ -0,0 +1,121 @@
+#include "kllvm/ast/AST.h"
+#include "kllvm/parser/KOREParser.h"
+#include "llvm/Support/CommandLine.h"
+
+#include <dlfcn.h>
+#include <iostream>
+#include <stdlib.h>
+
+#include "runtime/header.h"
+
+using namespace llvm;
+using namespace kllvm;
+
+struct Location {
+  char *filename;
```

Comment:
Don't store the raw `char *` in the struct - see comment below. This is a use-after-free waiting to happen!